src/main/java/net/minecraftforge/unification/UnificationManager.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package net.minecraftforge.unification;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import com.google.common.base.Preconditions;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.oredict.OreDictionary;
+
+/**
+ * The unification manager is the main access point for getting instances of:
+ * {@link UnifiedMaterial}, {@link UnifiedForm}, and {@link Unifier}.
+ */
+public final class UnificationManager
+{
+    private static final Map<String, UnifiedMaterial> MATERIALS = new HashMap<>();
+    private static final Map<String, UnifiedForm> FORMS = new HashMap<>();
```

Comment:
Hmm, these are not constants, so they should not be capitalized.

Reply:
I would argue that as you cannot change them without reflection, they are

Reply:
It's a mutable map, so yes, they are not constant. If you want them to be immutable (they should be, right?), use an `ImmutableMap`.

Reply:
They are just maps to keep registered materials and forms, and are never frozen.

Reply:
It's not clear to me what the convention is. I figure if it's `static final` then it should be all capitals, and that seems to be how most of the Minecraft and Forge codebase is done. If you can point to an example of where it's different then I'll give this more consideration, but looking online at old style guides I found a few  conflicting ideas that could support either style here.

Reply:
Ideally only true constants are capitalized. But this is not the case in the Forge/MCP style indeed. Oh well 😿